### ISSUE TITLE
data.frame(rS$rowSums, 1) :  arguments imply differing number of rows: 0, 1 in addBgdPeaks

### ISSUE THREAD

**Issue by Zepeng-Mu** [GreenleafLab/ArchR#742]
Hi,
I encountered this error in `addBgdPeaks`:
```r
projPBMC1 <- addBgdPeaks(projPBMC1)
```
```
Identifying Background Peaks!
Error in data.frame(rS$rowSums, 1) : 
  arguments imply differing number of rows: 0, 1
```
I'm not sure what caused this error message, and it's very hard for me to trouble shoot. I followed ArchR tutorial for my analysis and didn't get any other errors up until this step.

Thanks!

**Comment by rcorces** [comment]
I'm not really sure how to help. You'll need to provide a lot more information. My advice would be to try to step through the code with your data. Its probably something specific to your data. If you can recapitulate the error with the tutorial data, that will be helpful.

**Comment by Zepeng-Mu** [comment]
Hi,
I looked around the ArchR scripts and realized the error is due to an empty `rS` data frame, which is calculated from `PeakMatrix`. Then I found that I skipped one step so that I do not have a PeakMatrix but only a PeakSet. This becomes clear when I tried to use the matrix get-er function or the `getAvailableMatrices` function.

Somehow, `addBgdPeaks()`, specifically `.getRowSums()` does not check whether the `useMatrix` is available and throws an error if not, instead an empty `rS` is returned. Other functions like `getMatrixFromProject` do check if the matrix is present, so the error message is more informative if the matrix is not there.

I should have followed the tutorial more carefully, but I added a peak matrix using the peak set and everything is fine now.
Thanks!
